ethcore/src/client/client.rs

```diff
@@ -1318,6 +1318,25 @@ impl BlockChainClient for Client {
 		self.do_virtual_call(&env_info, &mut state, &t, analytics)
 	}
 
+	fn replay_block_transactions(&self, block: BlockId, analytics: CallAnalytics) -> Result<Vec<Executed>, CallError> {
+		let mut replays = Vec::new();
+		let mut env_info = self.env_info(block).ok_or(CallError::StatePruned)?;
+		let body = self.block_body(block).ok_or(CallError::StatePruned)?;
+		let mut state = self.state_at_beginning(block).ok_or(CallError::StatePruned)?;
+		let txs = body.transactions();
+
+		const PROOF: &'static str = "Transactions fetched from blockchain; blockchain transactions are valid; qed";
+		const EXECUTE_PROOF: &'static str = "Transaction replayed; qed";
+		for t in txs {
+			let t = SignedTransaction::new(t).expect(PROOF);
+			let x = self.do_virtual_call(&env_info, &mut state, &t, analytics).expect(EXECUTE_PROOF);
```

Comment:
You can just do `self.do_virtual_call(..)?` here. But you are right that the call should never error here, so I'm fine with expect as well.